src/plugins/kibana_react/public/core/memory.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Core } from './types';
+import { createInMemoryUiSettingsService } from './memory/ui_settings';
+
+export const createInMemoryCore = (): Core => {
```

Comment:
What problem do the inmemory services and the uisettings in memory service solve?
What's the use case?

Reply:
I can see how an in-memory instance of core services would be useful for testing (because you could actually interact with it, vs just having mocked functions that do nothing). But I think it warrants further discussion, and I don't think it falls under the scope of this PR.

In the interest of getting this merged soon, I'd recommend we pull it out and discuss this topic in a separate thread.

Reply:
We discussed with Peter in-memory implementation for Core before, I added a sample here as example for discussion. The best place for such implementation is in the Core though, next to the real implementation.

It makes testing much nicer and more correct. As well as might improve the real implementation as well if you write it together with in-memory implementation.

You can see now I'm using mocks and the test cases test less and are harder to understand.